subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -555,8 +612,12 @@ uint8_t ll_adv_aux_ad_data_set(uint8_t handle, uint8_t op, uint8_t frag_pref,
 		pdu_chain->len = sec_len + ad_len_overflow + len;
 
 		/* Fill AD Data in chain PDU */
-		(void)memcpy(dptr_chain, data, len);
-
+		if (ad_len_overflow != 0) {
+			(void)memcpy(dptr_chain, data, ad_len_overflow);
+		}
+		if (ad_len_chain != 0) {
+			(void)memcpy(dptr_chain, &data[ad_len + ad_len_overflow], len);
```

Comment:
Should this have been
```suggestion
			(void)memcpy(dptr_chain, &data[ad_len + ad_len_overflow], ad_len_chain);
```

Reply:
fixed